pkg/controller/migcluster/validation.go

```diff
@@ -335,3 +345,44 @@ func (r *ReconcileMigCluster) validateSaTokenPrivileges(cluster *migapi.MigClust
 	}
 	return nil
 }
+
+// validate operator version.
+func (r ReconcileMigCluster) validateOperatorVersionMatchesHost(cluster *migapi.MigCluster) error {
+	if cluster.Spec.IsHostCluster {
+		return nil
+	}
+	if cluster.Status.HasCriticalCondition() {
+		return nil
+	}
+
+	client, err := cluster.GetClient(r)
```

Comment:
I think naming this `clusterClient` may be helpful here to differentiate the reconciler client (r) from a MigCluster client.